addons/mail/static/src/models/channel_member_list_view.js

```diff
@@ -1,32 +1,63 @@
 /** @odoo-module **/
 
 import { registerModel } from '@mail/model/model_core';
-import { one } from '@mail/model/model_field';
-import { clear, replace } from '@mail/model/model_field_command';
+import { attr, one } from '@mail/model/model_field';
+import { clear, replace, insert} from '@mail/model/model_field_command';
 
 registerModel({
     name: 'ChannelMemberListView',
     identifyingFields: [['chatWindowOwner', 'threadViewOwner']],
+    lifecycleHooks: {
+        _created() {
+            this.fetchChannelMembers();
+        }
+    },
     recordMethods: {
+        /**
+         * Handles click on the "load more members" button.
+         */
+        async onClickLoadMoreMembers() {
+            this.fetchChannelMembers();
+        },
+        async fetchChannelMembers() {
+            const channelMembersList = await this.messaging.rpc({
+                model: 'mail.channel',
+                method: 'load_more_members',
+                args: [[this.channel.id]],
+                kwargs: {
+                    'known_member_ids': this.channel.channelMembers.map(channelMember => channelMember.partner.id),
+                },
+            });
+            if (!this.channel) {
+                return;
+            }
+            this.channel.update({ memberCount: channelMembersList.count });
+            for (const partner of channelMembersList.partners) {
+                const relation = { channel: replace(this.channel) };
+                relation.partner = insert(partner);
+                this.channel.update({ channelMembers: insert(relation) });
```

Comment:
This is basically this:
```js
this.channel.update({
    channelMembers: insert({
        channel: replace(this.channel),
        partner: replace(partner),
    })
});
```
Which is much easier to read that way.

_(yes, it's `replace(partner)` instead of `insert(partner)`)_

Reply:
`channel: replace(this.channel),` is not needed either, as inverse are automatically set when writing on relation